tests/database/test_database.py

```diff
@@ -0,0 +1,150 @@
+        """Test that DuplicateRecordError is raised when the email is already in use."""
+        with pytest.raises(DuplicateRecordError):
+            access_db.add_user("testuser", "testuser@example.com", "hashed_password")
+
+
+class TestRemoveUser:
+    """
+    Test removing a user from the database.
+    Includes tests for both success and failure cases.
+    """
+
+    def test_remove_user_success(self, access_db, create_user):
+        """Test that a user is successfully removed from the database."""
+
+        # Check if the user exists before removal
+        user_exists = access_db.get_id_from_email("testuser@example.com")
+        assert user_exists == 1
+
+        # Remove the user with ID 1
+        access_db.remove_user(user_id=1)
+
+        # Try to retrieve the user's ID after removal, expecting an exception
+        with pytest.raises(GetIdFromEmailError):
+            access_db.get_id_from_email("testuser@example.com")
+
+    def test_remove_user_not_found(self, access_db):
+        """Test that trying to remove a non-existing user raises GetUserFromIdError."""
+
+        # Try to remove a user with ID 1 (non-existent user)
+        with pytest.raises(GetUserFromIdError):
+            access_db.remove_user(user_id=1)
+
+
+class TestGetUserInfo:
+    """
+    Test User information retrieving.
+    Includes separate tests for each field type and errors.
+    """
+
+    def test_get_password_success(self, access_db, create_user):
+        """Test that the correct password hash is retrieved for an existing user."""
+        password_hash = access_db.get_user_info(user_id=1, field="password_hash")
+        # password would be hashed in the actual database
+        # it is not in tests for ease of use
+        assert password_hash == "hashed_password"
+
+    def test_get_username_success(self, access_db, create_user):
+        """Test that the correct username is retrieved for an existing user."""
+        username = access_db.get_user_info(user_id=1, field="username")
+        assert username == "testuser"
+
+    def test_get_email_success(self, access_db, create_user):
+        """Test that the correct email is retrieved for an existing user."""
+        email = access_db.get_user_info(user_id=1, field="email")
+        assert email == "testuser@example.com"
+
+    def test_get_date_created_success(self, access_db, create_user):
+        """Test that the correct creation date is retrieved for an existing user."""
+        created_at = access_db.get_user_info(user_id=1, field="created_at")
+        # Checks a datetime is returned
+        assert isinstance(created_at, datetime)
+
+    def test_get_user_not_found(self, access_db):
+        """Test that a GetUserFromIdError is raised when the user does not exist."""
+        with pytest.raises(GetUserFromIdError, match="User with ID 1 not found"):
+            access_db.get_user_info(user_id=1, field="password_hash")
```

Comment:
```suggestion
    def test_get_user_not_found(self, access_db, create_user):
        """Test that a GetUserFromIdError is raised when the user does not exist."""
        with pytest.raises(GetUserFromIdError, match="User with ID 2 not found"):
            access_db.get_user_info(user_id=2, field="password_hash")
```
similar to earlier comment

Reply:
Resolved in [d91ce60](https://github.com/rowan04/L5-SE-CW2-QualityPlan/pull/14/commits/d91ce60cb4fba350a53d8e52a5b499b8a095c5d1)